test/connectObservable.test.tsx

```diff
@@ -35,6 +35,35 @@ describe("connectObservable", () => {
     expect(result.current).toEqual(1)
   })
 
+  it("Only update when the previous and current update are distinct according to the comparator function", async () => {
+    interface TestUpdate {
+      value: number
+      valueToIgnore: string
+    }
+    const stream$ = new BehaviorSubject<TestUpdate>({
+      value: 0,
+      valueToIgnore: "A",
+    })
+
+    const compare = (a: TestUpdate, b: TestUpdate) => a.value === b.value
+    const [useLatestValue] = connectObservable(stream$, { compare })
+
+    const { result, rerender } = renderHook(() => useLatestValue())
+    expect(result.current.valueToIgnore).toEqual("A")
+    expect(result.current.value).toEqual(0)
+
+    stream$.next({ value: 0, valueToIgnore: "B" })
+    rerender()
+    //should not update to the latest value in the stream
+    expect(result.current.valueToIgnore).toEqual("A")
+    expect(result.current.value).toEqual(0)
+
+    stream$.next({ value: 1, valueToIgnore: "B" })
+    rerender()
```

Comment:
Small detail - I think `act` would be more semantic instead of `rerender` -

e.g.
```ts
act(() => {
    stream$.next({ value: 1, valueToIgnore: "B" })
})
```

Reply:
Now that I think of, in fact, it would be awesome if we can test that the component actually doesn't rerender if `valueToIgnore` changes and `value` doesn't. It's something that would add more value.